watcher/watcher.go

```diff
@@ -0,0 +1,156 @@
+package watcher
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Tonkpils/snag/exchange"
+	fsn "gopkg.in/fsnotify.v1"
+)
+
+const rebuild = "rebuild"
+
+type Watcher interface {
+	Watch(string) error
+}
+
+type FSWatcher struct {
+	ex           exchange.SendListener
+	fsn          *fsn.Watcher
+	done         chan struct{}
+	mtimes       map[string]time.Time
+	watching     map[string]struct{}
+	watchDir     string
+	ignoredItems []string
+}
+
+func New(ex exchange.SendListener, ignoredItems []string) (Watcher, error) {
```

Comment:
Wouldn't passing a channel replace the exchange concept? I'm not opposed to the creator managing the delivery of message but my fear is that it will get convoluted when we add in the UI component.